### PR TITLE
[On pause] Migration of International > locations> Countries listing

### PR DESCRIPTION
**Pull request by RaimondasSapola**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Migration of /international/locations/countries list page row and bulk actions and countries options
| Type?         | refacto
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | https://github.com/PrestaShop/PrestaShop/issues/10604 
| How to test?  | Countries page should remain unchanged

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/15096)

### CODE REVIEW

**Comment by RaimondasSapola** [comment]
Countries page migration needs bulk action assign to new zone. Waiting for https://github.com/PrestaShop/PrestaShop/pull/13988/files#diff-0b98e176ebce1408aab72e8bf1a6a60dR43 for reusable modal element.

**Comment by matks** [comment]
@RaimondasSapola https://github.com/PrestaShop/PrestaShop/pull/13988 has been merged 😉

File: src/PrestaShopBundle/Form/Admin/Type/ZoneChoiceType.php [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                    ['--' => ''],
+                    $this->zonesChoiceProvider->getChoices([
+                        'active' => $options['active'],
+                        'active_first' => $options['active_first'],
+                    ])
+                );
+
+                return $choices;
+            }
+        );
```
Comment (tdavidsonas88) [inline src/PrestaShopBundle/Form/Admin/Type/ZoneChoiceType.php]: already refactored in https://github.com/PrestaShop/PrestaShop/pull/15071 so would be nice to rebase correctly after merging #15071

File: src/PrestaShopBundle/Form/Admin/Type/ZoneChoiceType.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        // Set normalizer enables to use closure for choice generation with options
+        $resolver->setNormalizer(
+            'choices', function (Options $options) {
```
Comment (tdavidsonas88) [inline src/PrestaShopBundle/Form/Admin/Type/ZoneChoiceType.php:60]: was already refactored in #15071 so will need rebase after #15071 is merged

**Comment by tdavidsonas88** [comment]
Recommend to firstly merge #15071  and only then to rebase and merge this branch.

File: admin-dev/themes/new-theme/.webpack/common.js [hunk 3]
```diff
@@ -98,6 +98,7 @@ module.exports = {
     translation_settings: './js/pages/translation-settings',
     translations: './js/app/pages/translations',
     webservice: './js/pages/webservice',
+    country: './js/pages/country',
```
Comment (PierreRambaud) [inline admin-dev/themes/new-theme/.webpack/common.js]: Please sort key

**Comment by NeOMakinG** [comment]
Also, don't forget to update license header as your PR has been created before the global update :)

File: src/PrestaShopBundle/Controller/Admin/Improve/International/CountryController.php [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+     */
+    public function deleteBulkAction(Request $request): RedirectResponse
+    {
+        $countryIds = $this->getBulkCountriesFromRequest($request);
+
+        try {
+            $this->getCommandBus()->handle(new BulkDeleteCountriesCommand($countryIds));
+            $this->addFlash(
+                'success',
+                $this->trans('Successful deletion.', 'Admin.Notifications.Success')
```
Comment (Julievrz) [inline src/PrestaShopBundle/Controller/Admin/Improve/International/CountryController.php:100]:
```suggestion
                $this->trans('Successful deletion', 'Admin.Notifications.Success')
```

File: src/PrestaShopBundle/Controller/Admin/Improve/International/CountryController.php [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+     */
+    private function getErrorMessages(): array
+    {
+        return [
+            DeleteCountryException::class => $this->trans(
+                'An error occurred while deleting the object.',
+                'Admin.Notifications.Error'
+            ),
+            CountryNotFoundException::class => $this->trans(
+                'The object cannot be loaded (or found)',
```
Comment (Julievrz) [inline src/PrestaShopBundle/Controller/Admin/Improve/International/CountryController.php:257]:
```suggestion
                'The object cannot be loaded (or found).',
```

File: src/PrestaShopBundle/Controller/Admin/Improve/International/CountryController.php [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+                    'Admin.Notifications.Error'
+                ),
+                UpdateCountryException::FAILED_UPDATE_STATUS => $this->trans(
+                    'An error occurred while updating the status.',
+                    'Admin.Notifications.Error'
+                ),
+            ],
+            CountryConstraintException::class => [
+                CountryConstraintException::INVALID_ID => $this->trans(
+                    'The object cannot be loaded (the identifier is missing or invalid)',
```
Comment (Julievrz) [inline src/PrestaShopBundle/Controller/Admin/Improve/International/CountryController.php:272]:
```suggestion
                    'The object cannot be loaded (the identifier is missing or invalid).',
```

File: src/PrestaShopBundle/Resources/views/Admin/Improve/International/Locations/Country/Blocks/country_options.html.twig [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+{{ form_start(countryOptionsForm, {'action': path('admin_countries_save_options')}) }}
+<div class="card">
+  <h3 class="card-header">
+    {{ 'Country options'|trans({}, 'Admin.International.Feature') }}
+  </h3>
+  <div class="card-block row">
+    <div class="card-text">
+      <div class="form-group row">
+        <label class="form-control-label" for="{{ countryOptionsForm.options.restrict_delivery_countries.vars.id }}">
+          {{ 'Restrict country selections in front office to those covered by active carriers'|trans({}, 'Admin.International.Help') }}
```
Comment (Julievrz) [inline src/PrestaShopBundle/Resources/views/Admin/Improve/International/Locations/Country/Blocks/country_options.html.twig:37]:
```suggestion
          {{ 'Restrict country selections in front office to those covered by active carriers.'|trans({}, 'Admin.International.Help') }}
```

**Comment by matks** [comment]
@RaimondasSapola  I close this PR as it currently appears in our backlog of PRs and until I (or someone else) find some time to finish it, it won't move 😞  .

But when someone can finally resume work on this topic, she/he will be able to use this PR as a base.

**Pull request closed without merging**